### ISSUE TITLE
New 1.8 Blocks are not supported fully

### ISSUE THREAD

**Issue by veteranmina** [mcMMO-Dev/mcMMO#2466]
When mining, double drops do not give you the same item that you mined, but instead regular stone.

I am running the dev Build #31 (Feb 17, 2015 11:12:38 AM

**Comment by TfT-02** [comment]
Which version of Spigot are you using?

**Comment by veteranmina** [comment]
Sorry I had not replied for a long time, just got married and this wasn't really on my radar till now.

Current version we are using is  git-Spigot-e497721-96d09e3 (MC: 1.8) (Implementing API version 1.8-R0.1-SNAPSHOT)

**Comment by veteranmina** [comment]
To clarify the items being mined, they are the new blocks that were implemented in 1.8. That I can tell pre-1.8 blocks still work.

**Comment by veteranmina** [comment]
Any updates on this?

**Comment by TfT-02** [comment]
Congratulations on getting married! I'm quite busy as well, so please excuse me for my late reply.
As for this issue, we are using API to grab the drops from a certain block. Which means that normally, Spigot should tell mcMMO to drop the correct item when we ask which drop comes from one of the new 1.8 blocks. We'll check what is going on, might be a bug on their side though.

**Comment by astemus** [comment]
I have only noticed this behavior when using silk touch to mine the new stone types.  Using a pickaxe without silk touch gives the expected stone types.

I'm not sure, but I think there is already an open issue for this behavior?

**Comment by veteranmina** [comment]
Now to come to think of it astemus, my pick is silk touch enchanted. I will try a non-enchanted pick...

**Comment by astemus** [comment]
Here's a fix for the new stone double drops with silk touch. It should replace case STONE:  in the handleSilkTouchDrops switch in Mining.java.  It just bypasses the silk touch check for any stone that isn't smooth stone, since the new stone types have no silk drops.

            case STONE:
                if (blockState.getData().getData() == 0x0) {
                    Misc.dropItem(blockState.getLocation(), new ItemStack(blockType));
                }
                else {
                    handleMiningDrops(blockState);
                }
                return;

I don't have a fork active for this or I would submit a pull request...  I figured this would have been fixed by one of the devs already.  I'll set up a fork and submit it if I don't hear anything in this thread about it in the next few days...
